### PR TITLE
first pass at writing docs that will match the v2 branches

### PR DESCRIPTION
**Pull request by joewagner**
WIP updates to match the changes coming with all of the `v2` branches

### CODE REVIEW

**Comment by joewagner** [comment]
Hey @carsonfarmer, Just wanna check that this kind of PR for docs works for you?  I don't have access to gitbook, which is totally fine, so I was planning to make PRs via github.  I have another set of changes to make related to the separation of read and write queries, but wanna make sure I'm doing it right first.
Thanks!

File: building/javascript-sdk.md [hunk 1]
```diff
@@ -56,17 +50,19 @@ import { connect } from "@tableland/sdk";
...
+  `mytable` // optional perfix that will be added to "queryable name"
 );
 
-// `queryableName` will be the table name you chose with the
-// table id separated by and underscore 
+// `queryableName` will be the prefix, if you chose one, with the chain id and
+// table id separated by underscores 
 const queryableName = createRes.name;
-console.log(queryableName); // e.g. mytable_1
+console.log(queryableName); // e.g. mytable_4_1
```
Comment (carsonfarmer) [inline building/javascript-sdk.md:61]: I can't do a "suggestion" on delete lines for some reason, so here's the suggestion:

```
// `name` will be the prefix, if supplied, with the chain id and
// table id separated by underscores.
console.log(createRes.name)
```

Now that name is much simpler to reason about, I don't think we need to have a separate variable for it here. That makes the example simpler, and highlights that they'll get the name from the response, and they should just use that.

Comment (carsonfarmer) [reply building/javascript-sdk.md:61]: The above obviously implies that we'll need to change all use of queryableName => createRes.name. The alternative is to simply destructure the create response into name and whatever else we want/need in the example for a cleaner demo?

File: building/javascript-sdk.md [hunk 2]
```diff
@@ -86,17 +82,19 @@ const signer = wallet.connect(provider);
...
 const createRes = await tbl.create(
-  `CREATE TABLE mytable (name text, id int, primary key (id));`
+  4,  // Chain Id, note that since we connected to rinkeby testnet above this **MUST** be 4
+  `name text, id int, primary key (id);`, // table schema definition
+  `mytable` // optional perfix that will be added to "queryable name"
 );
 
-// `queryableName` will be the table name you chose with the
-// table id separated by and underscore 
+// `queryableName` will be the prefix, if you chose one, with the chain id and
```
Comment (carsonfarmer) [inline building/javascript-sdk.md:90]: Same deal here, keep it simple and just log the name directly.

File: building/javascript-sdk.md [hunk 3]
```diff
@@ -113,17 +111,19 @@ const provider = new ethers.providers.AlchemyProvider("rinkeby", "myapikey");
...
+// table id separated by underscores 
 const queryableName = createRes.name;
```
Comment (carsonfarmer) [inline building/javascript-sdk.md:121]: you get the idea

File: building/javascript-sdk.md [hunk 4]
```diff
@@ -136,17 +136,19 @@ import { connect } from "https://cdn.skypack.dev/@tableland/sdk";
...
-console.log(queryableName); // e.g. mytable_1
+console.log(queryableName); // e.g. mytable_4_1
```
Comment (carsonfarmer) [inline building/javascript-sdk.md:147]: last one

File: building/javascript-sdk.md [hunk 5]
```diff
@@ -201,7 +203,13 @@ const tbl = await tl.connect({ signer, network: "testnet" });
 
 ### Creating Tables
 
-Like most relational database systems, Tableland requires the user to create tables for storing, querying, and relating data. This is done via the `create` function. The `create` function takes a plain SQL statement string. Most valid SQL _constraints_ are supported, and the following data types are currently [supported](https://github.com/textileio/go-tableland/blob/main/pkg/parsing/query\_validator.go):
+Like most relational database systems, Tableland requires the user to create tables for storing, querying, and relating data. This is done via the `create` function. The `create` function takes 3 arguments: 
+1. The chainId of the chain you specified while connecting.
```
Comment (carsonfarmer) [inline building/javascript-sdk.md:207]: When we make this optional, let's also document that it is optional here. Note also that that means we have to change the structure of the function signature. We can't have an optional as the first argument.

File: building/javascript-sdk.md [hunk 6]
```diff
@@ -201,7 +203,13 @@ const tbl = await tl.connect({ signer, network: "testnet" });
...
+2. The column names, datatypes, and constraints that would go inside the parenthesis of a full `CREATE TABLE` statement.
```
Comment (carsonfarmer) [inline building/javascript-sdk.md:208]: Maybe we should provide an example here?

```
`(
  id int,
  name string,
  value real
)`
```

File: building/javascript-sdk.md [hunk 7]
```diff
@@ -201,7 +203,13 @@ const tbl = await tl.connect({ signer, network: "testnet" });
...
+3. An optional "prefix" to the universally unique Tableland table name.
+
+All tables in Tableland are unique, and the way uniqueness is ensured is that each table is named by combining the chainId of the chain the table is minted on, and the id of the ERC-721 token that represents the table.  If the prefix is specified it is added to to start of the table name.  As such the prefix must follow the rules SQL table naming, e.g it cannot start with a number and the dash character cannot be used.  For more on rules of SQL table names see [Naming Syntax](https://www.postgresql.org/docs/7.0/syntax525.htm)
```
Comment (carsonfarmer) [inline building/javascript-sdk.md:211]:
```suggestion
All tables in Tableland are unique, and the way uniqueness is ensured is that each table is named by combining the chainId of the chain the table is minted on, and the id of the ERC-721 token that represents the table. If the prefix is specified it is added to the start of the table name. As such the prefix must follow standard SQL table naming conventions, i.e. the prefix cannot start with a number, and only ASCII characters, numbers, and underscores may be used. For more on the rules of SQL table names see (for example) [Naming Syntax](https://www.postgresql.org/docs/7.0/syntax525.htm)
```

Also no double spaces after a period!

File: building/javascript-sdk.md [hunk 8]
```diff
@@ -201,7 +203,13 @@ const tbl = await tl.connect({ signer, network: "testnet" });
...
+Most valid SQL _constraints_ are supported, and the following data types are currently [supported](https://github.com/textileio/go-tableland/blob/main/pkg/parsing/query\_validator.go):
```
Comment (carsonfarmer) [inline building/javascript-sdk.md:212]: Actually, we should update this to point to the tableland spec docs now: 

```suggestion
Most valid SQL _constraints_ are supported, and the following data types are currently [supported](https://textile.notion.site/Data-Types-4831202597e04823998fe7addc275fea):
```

File: building/javascript-sdk.md [hunk 9]
```diff
@@ -201,7 +203,13 @@ const tbl = await tl.connect({ signer, network: "testnet" });
...
 
 * `int2`, `int4`, `int8`, `serial`, `bigserial`
```
Comment (carsonfarmer) [inline building/javascript-sdk.md:214]:
```suggestion
* `int`, `real`, `text`, `blob`, `any`
```

File: building/javascript-sdk.md [hunk 10]
```diff
@@ -201,7 +203,13 @@ const tbl = await tl.connect({ signer, network: "testnet" });
...
 * `text`, `uri`, `varchar`, `bpchar`
```
Comment (carsonfarmer) [inline building/javascript-sdk.md:215]:
```suggestion

See also the [Data Types docs](https://textile.notion.site/Data-Types-4831202597e04823998fe7addc275fea) for details on representing other common SQL data types using the above core types.
```

File: building/javascript-sdk.md [hunk 11]
```diff
@@ -217,18 +225,20 @@ The above list is tentative and incomplete; the accepted types are still not wel
...
 const createRes = await tbl.create(
-  "CREATE TABLE mytable (name text, id int, primary key (id));"
+  4,  // Chain Id, note that since we connected to rinkeby testnet above this **MUST** be 4
+  `name text, id int, primary key (id);`, // table schema definition
+  `mytable` // optional perfix that will be added to "queryable name"
 );
 
-// `queryableName` will be the table name you chose with the
-// table id separated by and underscore 
+// `queryableName` will be the prefix, if you chose one, with the chain id and
```
Comment (carsonfarmer) [inline building/javascript-sdk.md:233]: same again, wow we use this example a lot!

File: building/javascript-sdk.md [hunk 12]
```diff
@@ -217,18 +225,20 @@ The above list is tentative and incomplete; the accepted types are still not wel
...
-Queries and updates will use the table id with the table name from the create statement by combining them with the underscore character as a separator.  This results in the "queryable name".  The code snippets above show examples of this where the first table in Tableland has the name `mytable`, resulting in a queryable name of `mytable_1`.
+Creating a table generates a unique table identifier (table id). The table id is unique within the specified chain and can be used to reference the table for queries and updates after it has been created.  This table id is the tokenId from the Tableland Registry Smart Contract which means you can view your table on any blockchain scanner associated with the chain your table was minted on, Here is an example of a Tableland rinkeby table viewed on [Opensea](https://testnets.opensea.io/assets/0x30867ad98a520287ccc28cde70fcf63e3cdb9c3c/723)
```
Comment (carsonfarmer) [inline building/javascript-sdk.md:240]:
```suggestion
Creating a table generates a unique table identifier (table id). The table id is unique within the specified chain and can be used to reference the table for queries and updates after it has been created.  This table id is the tokenId from the Tableland Registry Smart Contract which means you can view your table on any blockchain scanner associated with the chain your table was minted on. Here is an example of a Tableland rinkeby table viewed on [Opensea](https://testnets.opensea.io/assets/0x30867ad98a520287ccc28cde70fcf63e3cdb9c3c/723)
```

**Review by carsonfarmer — CHANGES_REQUESTED** [review]
Awesome! This is a great update on the SDK API given all the work you've done on the SDK! I'd made a few suggested changes, and where I couldn't, I've made comments. Minor nit-style updates really.

We'll have to also make some updates to the other sections, including the command line, the RPC endpoints, and more. We'll also eventually have to create updates for the GRANT and REVOKE APIs, and all the work on the Controller contract stuff! I know that is out of scope... just documenting here for posterity.

**Review by carsonfarmer — APPROVED** [review]
Sweet!

**Pull request merged**